tests/models/edx/test_base.py

```diff
@@ -0,0 +1,70 @@
+"""Tests for the base event model"""
+
+import pytest
+from pydantic.error_wrappers import ValidationError
+
+from tests.fixtures.edx.server import BaseEventFactory
+from tests.fixtures.logs import event_factory
+
+
+@pytest.mark.parametrize(
+    "kwargs",
+    [
+        {"ip": "51.255.9.27"},
+        {"ip": "8.8.8.8"},
+        {"ip": ""},
+        {"event_source": "server"},
+        {"agent": ""},
+        {"host": ""},
+        {"referer": ""},
+        {"referer": "https://www.fun-mooc.fr/"},
+        {"referer": "https://www.fun-mooc.fr/event?event_type=page_close"},
+        {"context_": {"user_id": ""}},
+        {"context_": {"user_id": None}},
+        {"context_": {"user_id": 213456}},
+        {"context_": {"org_id": "", "course_id": ""}},
+        {"context_": {"org_id": "org", "course_id": "course-v1:org+any+any"}},
+    ],
+)
+def test_valid_base_event(kwargs):
+    """Test that a valid base event does not raise ValidationError."""
+
+    try:
+        event_factory(BaseEventFactory, **kwargs)
+    except ValidationError:
+        pytest.fail(f"Valid base event with {kwargs} should not raise exceptions")
+
+
+REGEX = "course_id must match regex"
```

Comment:
Does not look like a regex to me :wink: 

```suggestion
INVALID_COURSE_ID_ERROR = "course_id must match regex"
```

or repeat the expected error message in your test parameters as the variable name is not that short :stuck_out_tongue:

Reply:
Yes, indeed it's not a regex :sweat_smile:
To be honest I tried to make pylint keep one test per line, but with the subfactory syntax and long error messages, pylint takes over. Updated)